pkg/apis/workflow/v1alpha1/types.go

```diff
@@ -167,6 +167,14 @@ type WorkflowSpec struct {
 
 	// HostAliases is an optional list of hosts and IPs that will be injected into the pod spec
 	HostAliases []apiv1.HostAlias `json:"hostAliases,omitempty"`
+
+	// This flag is for DAG logic. The DAG logic has a built-in "fail fast" feature to stop scheduling new steps,
+	// as soon as it detects that one of the DAG nodes is failed. Then it waits until all DAG nodes are completed
+	// before failing the DAG itself.
+	// The FailFast flag default is true,  if set to false, it will allow a DAG to run all branches of the DAG to
+	// completion (either success or failure), regardless of the failed outcomes of branches in the DAG.
+	// More info and example about this feature at https://github.com/argoproj/argo/issues/1442
+	FailFast *bool `json:"failFast,omitempty"`
```

Comment:
Since this is a DAG specific field, I think this may be more suitable the DAG template? Downside is that FailFast would have to be set for every DAG template, if you want the entire workflow to behave this way. On the other hand, having it at DAG template level means you can decide to have some DAG template fail fast, and others not.

Reply:
Agree.